src/client/client.c

```diff
@@ -161,6 +162,129 @@ void unit_list_unref(UnitList *unit_list) {
         free(unit_list);
 }
 
+static sd_bus_message *wait_for_job(Client *client, char *object_path) {
+        int r = 0;
+
+        assert(client->waiting_for_job == NULL);
+        client->waiting_for_job = object_path;
+
+        for (;;) {
+                /* Process requests */
+                r = sd_bus_process(client->api_bus, NULL);
+                if (r < 0) {
+                        fprintf(stderr, "Failed to process bus: %s\n", strerror(-r));
+                        return NULL;
+                }
+
+                /* Did we get the result? */
+                if (client->waiting_for_job_result != NULL) {
+                        return steal_pointer(&client->waiting_for_job_result);
+                }
+
+                if (r > 0) { /* request processed, try to process another one */
+                        continue;
+                }
+
+                /* Wait for the next request to process */
+                r = sd_bus_wait(client->api_bus, (uint64_t) -1);
+                if (r < 0) {
+                        fprintf(stderr, "Failed to wait on bus: %s\n", strerror(-r));
+                        return NULL;
+                }
+        }
+}
+
+static int match_job_removed_signal(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *error) {
+        const char *job_path = NULL, *result = NULL, *node = NULL, *unit = NULL;
+        uint32_t id = 0;
+        int r = 0;
+        Client *client = (Client *) userdata;
+
+        if (client->waiting_for_job == NULL) {
```

Comment:
If I understand `sd_bus_process()` correctly, this check here is not necessary since it is set in `wait_for_job` before starting to process anything.

Reply:
Yes, but only after calling `sd_bus_process` the callback would be called, right? Otherwise I am wondering how the signal callback is being run (e.g. in a separate thread). But I mean, checking for NULL here is not wrong so its just minor.

Reply:
Checking for NULL never made a program fail.
Not checking for NULL does it every second somewhere in the world :) (This is not scientific, but I assume so)

Reply:
As if some NULL issue is the most common error ;) 
Well ok, then we'll leave it as it is now.